v3/src/components/case-table/use-rows.ts

```diff
@@ -98,6 +98,7 @@ export const useRows = (data?: IDataSet) => {
             // update cache entries for each affected case
             const cases = (action as AddCasesAction | SetCaseValuesAction).args[0] || []
             cases.forEach(({ __id__ }) => rowCache.set(__id__, { __id__ }))
+            resetRowCache()
```

Comment:
This should not be necessary. The point of the prior code is to make sure that only the rows that change re-render, while this will force all rows to re-render. If the more efficient code didn't work as expected then we should figure out why and fix it.